drivers/video/arducam_mega.c

```diff
@@ -0,0 +1,1048 @@
+
+	return 0;
+}
+
+static int arducam_mega_dequeue(const struct device *dev, enum video_endpoint_id ep,
+				struct video_buffer **vbuf, k_timeout_t timeout)
+{
+	struct arducam_mega_data *data = dev->data;
+
+	if (ep != VIDEO_EP_OUT) {
+		return -EINVAL;
+	}
+
+	*vbuf = k_fifo_get(&data->fifo_out, timeout);
+
+	LOG_DBG("dequeue buffer %p", (*vbuf)->buffer);
+
+	if (*vbuf == NULL) {
+		return -EAGAIN;
+	}
+
+	return 0;
+}
+
+static int arducam_mega_get_caps(const struct device *dev, enum video_endpoint_id ep,
+				 struct video_caps *caps)
+{
+	caps->format_caps = fmts;
+	return 0;
+}
+
+static int arducam_mega_set_ctrl(const struct device *dev, unsigned int cid, void *value)
+{
+	int ret = 0;
+
+	switch (cid) {
+	case VIDEO_CID_CAMERA_EXPOSURE_AUTO:
+		ret |= arducam_mega_set_exposure_enable(dev, *(uint8_t *)value);
+		break;
+	case VIDEO_CID_CAMERA_EXPOSURE:
+		ret |= arducam_mega_set_exposure(dev, *(uint32_t *)value);
+		break;
+	case VIDEO_CID_CAMERA_GAIN_AUTO:
+		ret |= arducam_mega_set_gain_enable(dev, *(uint8_t *)value);
+		break;
+	case VIDEO_CID_CAMERA_GAIN:
+		ret |= arducam_mega_set_gain(dev, *(uint16_t *)value);
+		break;
+	case VIDEO_CID_CAMERA_BRIGHTNESS:
+		ret |= arducam_mega_set_brightness(dev, *(enum MEGA_BRIGHTNESS_LEVEL *)value);
+		break;
+	case VIDEO_CID_CAMERA_SATURATION:
+		ret |= arducam_mega_set_saturation(dev, *(enum MEGA_SATURATION_LEVEL *)value);
+		break;
+	case VIDEO_CID_CAMERA_WHITE_BAL_AUTO:
+		ret |= arducam_mega_set_white_bal_enable(dev, *(uint8_t *)value);
+		break;
+	case VIDEO_CID_CAMERA_WHITE_BAL:
+		ret |= arducam_mega_set_white_bal(dev, *(enum MEGA_WHITE_BALANCE *)value);
+		break;
+	case VIDEO_CID_CAMERA_CONTRAST:
+		ret |= arducam_mega_set_contrast(dev, *(enum MEGA_CONTRAST_LEVEL *)value);
+		break;
+	case VIDEO_CID_JPEG_COMPRESSION_QUALITY:
+		ret |= arducam_mega_set_JPEG_quality(dev, *(enum MEGA_IMAGE_QUALITY *)value);
+		break;
+	case VIDEO_CID_ARDUCAM_EV:
+		ret |= arducam_mega_set_EV(dev, *(enum MEGA_EV_LEVEL *)value);
+		break;
+	case VIDEO_CID_ARDUCAM_SHARPNESS:
+		ret |= arducam_mega_set_sharpness(dev, *(enum MEGA_SHARPNESS_LEVEL *)value);
+		break;
+	case VIDEO_CID_ARDUCAM_COLOR_FX:
+		ret |= arducam_mega_set_special_effects(dev, *(enum MEGA_COLOR_FX *)value);
+		break;
+	case VIDEO_CID_ARDUCAM_RESET:
+		ret |= arducam_mega_soft_reset(dev);
+		ret |= arducam_mega_check_connection(dev);
+		break;
+	case VIDEO_CID_ARDUCAM_LOWPOWER:
+		ret |= arducam_mega_set_lowpower_enable(dev, *(uint8_t *)value);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return ret;
+}
+
+int arducam_mega_get_info(const struct device *dev, struct arducam_mega_info *info)
+{
+	struct arducam_mega_data *drv_data = dev->data;
+
+	*info = (*drv_data->info);
+
+	return 0;
+}
+
+static int arducam_mega_get_ctrl(const struct device *dev, unsigned int cid, void *value)
+{
+	int ret = 0;
+
+	switch (cid) {
+	case VIDEO_CID_ARDUCAM_INFO:
+		ret |= arducam_mega_get_info(dev, (struct arducam_mega_info *)value);
```

Comment:
As seen earlier, there cannot be `|=` as this scrambles the return status.

It's probably good enough to copy-paste such kind of boilerplate everywhere `ret |=` is seen (which also needs to become `ret = `:

```c
if (ret != 0) {
	return ret;
}
```

Thank you for the changes!